### PR TITLE
PAINTROID-314 Refactor MainActivityNavigator to Kotlin

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
Refactored `MainActivityNavigator` to Kotlin.

Replaced the deprecated `ACTION_MEDIA_SCANNER_SCAN_FILE` intent with `MediaScannerConnections` (reference: https://stackoverflow.com/questions/60203353/action-media-scanner-scan-filestring-is-deprecated)

Removed `@JvmStatic` annotation which is no longer required, and refactored the following dialog classes:

1. AboutDialog
2. FeedbackDialog
3. ImportImageDialog
4. InfoDialog
5. LikeUsDialog
6. OverwriteDialog
7. SaveInformationDialog
8. ScaleImageOnLoadDialog

https://jira.catrob.at/browse/PAINTROID-314

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+
+    override fun startWelcomeActivity(@ActivityRequestCode requestCode: Int) {
+        val intent = Intent(mainActivity.applicationContext, WelcomeActivity::class.java)
+        intent.flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+        mainActivity.startActivityForResult(intent, requestCode)
+    }
+
+    override fun startShareImageActivity(bitmap: Bitmap?) {
+        val uri = FileIO.saveBitmapToCache(bitmap, mainActivity)
+        if (uri != null) {
```
Comment (gPathpp) [inline Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt:231]:
```suggestion
        val uri = FileIO.saveBitmapToCache(bitmap, mainActivity) ?: return
```

**Review by gPathpp — CHANGES_REQUESTED** [review]
LGTM 👍
Just one small change please :)

File: Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt (hunk 1 above)
Comment (CodeChamp-SS) [reply Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt]: done :)

**Review by gPathpp — APPROVED** [review]

**Comment by ldirry** [comment]
This doesn't seem to be related with this PR but there is currently a bug on develop.

If you save an image as png first and then save it another time to jpg, it will get saved as <image-name>.png with jpg compression. Same if you save it as .jpg first and the second time as .png.

Is this an easy fix, would be nice if we could include it then? Else the PR looks good to me :)

**Comment by wslany** [comment]
Please also resolve conflicts.

**Comment by CodeChamp-SS** [comment]
> Please also resolve conflicts.

sure, will resolve them soon, also need to look into the thing Lorenz mentioned :)

**Review by ldirry — APPROVED** [review]
This bug: https://jira.catrob.at/browse/PAINTROID-344 wasn't introduced with this PR and we don't know a quick fix yet. PR looks good to me :). Merged.

**Comment by ldirry** [comment]
ah damn, my bad, I didn't see that the `testSaveAndOverrideOraFile` was crashing with this PR, reverted the merge sry...  do you maybe know how to fix this testcase? :)

**Pull request merged**